airflow-core/src/airflow/api_fastapi/core_api/routes/ui/dags.py

```diff
@@ -191,3 +195,40 @@ def get_dags(
         total_entries=total_entries,
         dags=list(dag_runs_by_dag_id.values()),
     )
+
+
+@dags_router.get(
+    "/{dag_id}/latest_run",
+    responses=create_openapi_http_exception_doc([status.HTTP_404_NOT_FOUND]),
+    dependencies=[Depends(requires_access_dag(method="GET", access_entity=DagAccessEntity.RUN))],
+)
+def get_latest_run_ui(dag_id: str, session: SessionDep) -> list[DAGRunLightResponse]:
+    """
+    Get latest run.
+
+    This endpoint allows specifying `~` as the dag_id to retrieve Dag Runs for all DAGs.
+    """
+    if dag_id == "~":
+        raise HTTPException(
+            status.HTTP_400_BAD_REQUEST,
+            "`~` was supplied as dag_id, but querying multiple dags is not supported.",
```

Comment:
Wait, why even document that we can if we don't let you?

Reply:
and.... i'm just making sure that someone doesn't accidentally do it